Add support for openEuler 22.03

martin-g opened the pull request:
Adds logic to support building and testing on openEuler 22.03.

For the time being it uses urls to openEuler's OBS and Github repo to fetch ohpc-release.oe2203.rpm until this package is made available at http://obs.openhpc.community

Signed-off-by: Yikun Jiang <yikunkero@gmail.com>
Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

adrianreber commented:
I would prefer smaller steps. Please create a PR just to introduce the Github Actions CI part based on https://obs.openhpc.community/project/show/OpenHPC:3.0:Factory

martin-g committed:
Remove the CI related changes

They were moved to https://github.com/openhpc/ohpc/pull/1647

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
Done!
All the CI related changes have been moved to https://github.com/openhpc/ohpc/pull/1647!

Yikun commented on components/admin/powerman/SPECS/powerman.spec:
```diff
@@ -22,7 +22,7 @@ Group: %{PROJ_NAME}/admin
 Source0: https://github.com/chaos/%{pname}/releases/download/%{version}/%{pname}-%{version}.tar.gz
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1656

Yikun commented on components/dev-tools/autoconf/SPECS/autoconf.spec:
```diff
@@ -24,7 +24,7 @@ Source0:   https://ftp.gnu.org/gnu/autoconf/autoconf-%{version}.tar.gz
 Requires: m4
 
-%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel}
+%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1657

Yikun commented on components/dev-tools/automake/SPECS/automake.spec:
```diff
@@ -20,6 +20,7 @@ License:   GNU GPL
 Group:     %{PROJ_NAME}/dev-tools
 URL:       http://www.gnu.org/software/automake/
 Source0:   https://ftp.gnu.org/gnu/automake/automake-%{version}.tar.gz
+Patch0:    fix-automake-build-error.patch
```
out date

Yikun commented on components/dev-tools/automake/SPECS/automake.spec:
```diff
@@ -28,6 +29,11 @@ BuildRequires: autoconf%{PROJ_DELIM} >= 2.69
 
 BuildRequires: perl(Thread::Queue) make
 
+%if 0%{?openEuler}
```
Replaced by https://github.com/openhpc/ohpc/commit/2ab677cbaf5b44c1afa4ce462f0c20c71e4eed52 (L30)

Yikun commented on components/dev-tools/cmake/SPECS/cmake.spec:
```diff
@@ -31,7 +31,7 @@ BuildRequires:  xz-devel
 BuildRequires:  pkgconfig
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1657

Yikun commented on components/dev-tools/spack/SPECS/spack.spec:
```diff
@@ -42,7 +42,7 @@ Requires: zstd
 Requires: git
 Requires: curl
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1657

Yikun commented on components/mpi-families/libfabric/SPECS/libfabric.spec:
```diff
@@ -28,7 +28,7 @@ BuildRequires: gcc make
 BuildRequires: rdma-core-devel infiniband-diags-devel
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1659

Yikun commented on components/mpi-families/mvapich2/SPECS/mvapich2.spec:
```diff
@@ -61,7 +61,7 @@ Conflicts: %{pname}-%{compiler_family}%{PROJ_DELIM}
 BuildRequires: rdma-core-devel infiniband-diags-devel
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1659

Yikun commented on components/mpi-families/openmpi/SPECS/openmpi.spec:
```diff
@@ -68,7 +68,7 @@ BuildRequires:  libevent-devel
 %if 0%{with_ofi}
 BuildRequires:  libfabric%{PROJ_DELIM}
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1659

Yikun commented on components/parallel-libs/boost/SPECS/boost.spec:
```diff
@@ -33,7 +33,7 @@ BuildRequires:  python3-numpy-devel
 %endif
 %else
-# Assume RHEL/Fedora distro
+# Assume RHEL/Fedora/openEuler distro
```
https://github.com/openhpc/ohpc/pull/1676

Yikun commented on components/parallel-libs/mumps/SPECS/mumps.spec:
```diff
@@ -36,7 +36,7 @@ Patch0:         mumps-5.0.1-shared-mumps.patch
 Requires:       lmod%{PROJ_DELIM} >= 7.6.1
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1677

Yikun commented on components/parallel-libs/ptscotch/SPECS/ptscotch.spec:
```diff
@@ -31,7 +31,7 @@ Patch0:  scotch-6.0.4-destdir.patch
 BuildRequires:	flex bison make
 BuildRequires:  zlib-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1678

Yikun commented on components/parallel-libs/petsc/SPECS/petsc.spec:
```diff
@@ -38,7 +38,7 @@ BuildRequires:  valgrind-devel
 BuildRequires:  xz
 BuildRequires:  zlib-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1677

Yikun commented on components/io-libs/sionlib/SPECS/sionlib.spec:
```diff
@@ -27,7 +27,7 @@ Source0:   http://apps.fz-juelich.de/jsc/sionlib/download.php?version=%{version}
 
 # For pre-processor only:
-%if 0%{?rhel}
+%if 0%{?rhel} ||  0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1682

Yikun commented on components/io-libs/netcdf-fortran/SPECS/netcdf-fortran.spec:
```diff
@@ -28,7 +28,7 @@ Source0:	https://github.com/Unidata/netcdf-fortran/archive/v%{version}.tar.gz
 BuildRequires:  zlib-devel >= 1.2.5
 BuildRequires:  libxml2-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1681

Yikun commented on components/distro-packages/python-Cython/SPECS/python-Cython.spec:
```diff
@@ -21,7 +21,7 @@ License:        Apache-2.0
 Source0:        https://files.pythonhosted.org/packages/source/C/Cython/Cython-%{version}.tar.gz
 Source1:        python-Cython-rpmlintrc
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1658

Yikun commented on components/perf-tools/paraver/SPECS/paraver.spec:
```diff
@@ -27,7 +27,7 @@ Source0:       https://ftp.tools.bsc.es/wxparaver/wxparaver-%{version}-src.tar.b
 BuildRequires: bison
 BuildRequires: boost-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1684

Yikun commented on components/perf-tools/dimemas/SPECS/dimemas.spec:
```diff
@@ -28,7 +28,7 @@ Source0:	https://ftp.tools.bsc.es/dimemas/dimemas-%{version}-src.tar.bz2
 Requires:      boost-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
 BuildRequires: bison make
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1683

Yikun commented on components/rms/munge/SPECS/munge.spec:
```diff
@@ -114,7 +114,7 @@ touch "$RPM_BUILD_ROOT"/%{_localstatedir}/run/munge/munged.pid
 rm "$RPM_BUILD_ROOT"/etc/init.d/munge
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1685

Yikun commented on components/serial-libs/openblas/SPECS/openblas.spec:
```diff
@@ -29,6 +29,7 @@ Patch2:         openblas-noexecstack.patch
 Patch3:         fix-arm64-cpuid-return.patch
 ExclusiveArch:  %ix86 ia64 ppc ppc64 ppc64le x86_64 aarch64
 
+BuildRequires:  make ohpc-buildroot
```
https://github.com/openhpc/ohpc/pull/1686

Yikun commented on components/serial-libs/R/SPECS/R.spec:
```diff
@@ -44,14 +44,18 @@ BuildRequires:  xz-devel
 BuildRequires:  libcurl-devel
 BuildRequires:  tk-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1687

Yikun commented on components/serial-libs/scotch/SPECS/scotch.spec:
```diff
@@ -27,7 +27,7 @@ Source2:	%{pname}-rpmlintrc
 
 BuildRequires:	flex bison make
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
https://github.com/openhpc/ohpc/pull/1688

martin-g committed:
openEuler support

Add TODOs to replace openEuler specific urls with xyz.openhpc.community ones

Until openEuler is added to the officially supported ones we have to use
the custom OBS

Signed-off-by: Yikun Jiang <yikunkero@gmail.com>
Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>
